src/Index.js

```diff
@@ -923,6 +923,69 @@ Index.prototype.exists = function(callback) {
   });
 };
 
+Index.prototype.findObject = function(findCallback, requestOptions, callback) {
+  requestOptions = requestOptions === undefined ? {} : requestOptions;
```

Comment:
```suggestion
  requestOptions =
    typeof requestOptions === 'undefined' || typeof requestOptions === 'function'
      ? {}
      : requestOptions;
```

Reply:
What do you propose is having `overloading` here correct?

Still, I would rather make the `JS` client similar to every other client instead, as people will not use `callback` anyway. What do you think?

Reply:
in a way yes, mostly just in case people give the callback as the second argument, since that's consistent with the other optional arguments. If you want to make this method more like v4 that's fine